### PR TITLE
refactor: LocationManager provider

### PR DESCRIPTION
**Pull request by Dor-bl**
- Updated createLocationManagerMockProvider method to use getProviderProperties for API level 31 and above.
- Added runtime API level checks to ensure backward compatibility with API levels below 31.
- Removed dependency on LocationProvider in the calling method.
- Handled null cases for provider name and provider properties.
- Ensured compatibility and adherence to modern Android best practices.

### CODE REVIEW

**Comment by Dor-bl** [comment]
@mykola-mokhnach , I see the install dependencies failed.
Should I add the below elsewhere? 
`import android.location.provider.ProviderProperties;`

**Comment by KazuCocoa** [comment]
For the lint error, it can have `@SuppressLint("MissingPermission")` for now I think

File: app/src/main/java/io/appium/settings/LocationService.java [hunk 1]
```diff
@@ -184,23 +185,49 @@ private List<MockLocationProvider> createMockProviders(LocationManager locationM
...
+                    providerProperties.hasSatelliteRequirement(),
+                    providerProperties.hasCellRequirement(),
+                    providerProperties.hasMonetaryCost(),
+                    providerProperties.hasAltitudeSupport(),
+                    providerProperties.hasSpeedSupport(),
+                    providerProperties.hasBearingSupport(),
+                    providerProperties.getPowerUsage(),
+                    providerProperties.getAccuracy()
+            );
+        } else {
```
Comment (mykola-mokhnach) [inline app/src/main/java/io/appium/settings/LocationService.java:210]: else is redundant

**Review by KazuCocoa — APPROVED** [review]

**Comment by KazuCocoa** [comment]
It looks like `Method getAnimationScales = windowManagerClass.getDeclaredMethod("getAnimationScales");` may raise an error for target api level 31 as https://developer.android.com/guide/app-compatibility/restrictions-non-sdk-interfaces


It looks like the animation command usage is only for lower than API level 26 https://github.com/appium/appium-uiautomator2-driver/blob/4cd489aa9a282f412ee45e1366c86bf7e580c798/lib/driver.ts#L594-L597
Espresso actually uses this method but I think espresso server also can follow the `--no-window-animation`. Then we can completely drop this animation command in settings.


We could add an annotation for `@SuppressLint("SoonBlockedPrivateApi"` to ignore this error I believe. We disable https://developer.android.com/guide/app-compatibility/restrictions-non-sdk-interfaces with hidden policy settings command call, but some devices may not allow to apply the hidden policy. So... I think it would be nice to drop lower Android OS version first, and remove animation command via settings app, then apply this change to settings.

File: app/build.gradle [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 
 android {
-    compileSdk 30
+    compileSdk 31
 
     defaultConfig {
         minSdkVersion 21
         //noinspection ExpiredTargetSdkVersion
-        targetSdkVersion 30
+        targetSdkVersion 31
```
Comment (KazuCocoa) [inline app/build.gradle:9]: Do you think if we can keep this target SDK level 30? Compile SDK can be greater than 31, of course. The main thing is the animation stuff in https://github.com/appium/io.appium.settings/pull/174#issuecomment-2130875350

Comment (KazuCocoa) [reply app/build.gradle:9]: Please ignore my ^ comment.
I think with https://github.com/appium/appium-espresso-driver/pull/1007 , the animation stuff affect only lower than api level 26, so the `@SuppressLint("SoonBlockedPrivateApi"` should be sufficient as this change

**Comment by KazuCocoa** [comment]
lg, but please wait for https://github.com/appium/appium-espresso-driver/pull/1007 so that we can safely apply the animation related change for lower os versions only

**Pull request merged**